akka-projection-r2dbc/src/main/scala/akka/projection/r2dbc/internal/R2dbcOffsetStore.scala

```diff
@@ -301,12 +302,36 @@ private[projection] class R2dbcOffsetStore(
       clearInflight()
       if (newState == State.empty) {
         None
+      } else if (moreThanOneProjectionKey(recordsWithKey)) {
+        // When downscaling projection instances (changing slice distribution) there
+        // is a possibility that one of the previous projection instances was further behind than the backtracking
+        // window, which would cause missed events if we started from latest. In that case we use the latest
+        // offset of the earliest slice
+        val latestBySlice = newState.latestBySlice
+        val earliest = latestBySlice.minBy(_.timestamp).timestamp
+        Some(TimestampOffset(earliest, Map.empty))
```

Comment:
For the latest offset, the seen map is populated. When I was experimenting with the earliest or latest-by-split offsets, I also ended up creating the seen map from `byPid` for the selected offset here. Shouldn't affect behaviour when eventually deduplicated, but I think at least the metrics were confused (inactive projections had growing consumer lag metrics).

Reply:
I don't think it's important to reconstruct it in this case, but I added the obvious from `earliest` record. https://github.com/akka/akka-projection/pull/1012/commits/4ab18c35d8c0ec4c88ef8eb4a75032ff164075fd